code/modules/modular_computers/file_system/programs/borg_self_monitor.dm

```diff
@@ -0,0 +1,143 @@
+
+	//DEBUG -- Cover, TRUE for locked
+	data["cover"] = "[borgo.locked? "LOCKED":"UNLOCKED"]"
+	//Ability to move. FAULT if lockdown wire is cut, DISABLED if borg locked, ENABLED otherwise
+	data["locomotion"] = "[borgo.wires.is_cut(WIRE_LOCKDOWN)?"FAULT":"[borgo.lockcharge?"DISABLED":"ENABLED"]"]"
+	//Module wire. FAULT if cut, NOMINAL otherwise
+	data["wireModule"] = "[borgo.wires.is_cut(WIRE_RESET_MODULE)?"FAULT":"NOMINAL"]"
+	//DEBUG -- Camera(net) wire. FAULT if cut (or no cameranet camera), DISABLED if pulse-disabled, NOMINAL otherwise
+	data["wireCamera"] = "[!borgo.builtInCamera || borgo.wires.is_cut(WIRE_CAMERA)?"FAULT":"[borgo.builtInCamera.can_use()?"NOMINAL":"DISABLED"]"]"
+	//AI wire. FAULT if wire is cut, CONNECTED if connected to AI, READY otherwise
+	data["wireAI"] = "[borgo.wires.is_cut(WIRE_AI)?"FAULT":"[borgo.connected_ai?"CONNECTED":"READY"]"]"
+	//Law sync wire. FAULT if cut, NOMINAL otherwise
+	data["wireLaw"] = "[borgo.wires.is_cut(WIRE_LAWSYNC)?"FAULT":"NOMINAL"]"
+
+	return data
+
+/datum/computer_file/program/borg_self_monitor/ui_static_data(mob/user)
+	var/list/data = list()
+	if(!iscyborg(user))
+		return data
+	var/mob/living/silicon/robot/borgo = user
+
+	data["Laws"] = borgo.laws.get_law_list(TRUE, TRUE, FALSE)
+	data["borgLog"] = tablet.borglog
+	data["borgUpgrades"] = borgo.upgrades
+	return data
+
+/datum/computer_file/program/borg_self_monitor/ui_act(action, params)
+	. = ..()
+	if(.)
+		return
+
+	var/mob/living/silicon/robot/borgo = tablet.borgo
+
+	switch(action)
+		if("coverunlock")
+			if(borgo.locked)
+				borgo.locked = FALSE
+				borgo.update_icons()
+				if(borgo.emagged)
+					borgo.logevent("ChÃ¥vÃis cover lock has been [borgo.locked ? "engaged" : "released"]") //"The cover interface glitches out for a split second"
+				else
+					borgo.logevent("Chassis cover lock has been [borgo.locked ? "engaged" : "released"]")
+
+		if("lawchannel")
+			borgo.set_autosay()
+
+		if("lawstate")
+			borgo.checklaws()
+
+		if("alertPower")
+			if(borgo.stat == CONSCIOUS)
+				if(!borgo.cell || !borgo.cell.charge)
+					borgo.visible_message("<span class='notice'>The power warning light on <span class='name'>[borgo]</span> flashes urgently.</span>", \
+						"You announce you are operating in low power mode.")
+					playsound(borgo, 'sound/machines/buzz-two.ogg', 50, FALSE)
+
+		if("toggleSensors")
+			borgo.toggle_sensors()
+
+		if("viewImage")
+			borgo.aicamera?.viewpictures(usr)
+
+		if("printImage")
+			var/obj/item/camera/siliconcam/robot_camera/borgcam = borgo.aicamera
+			borgcam?.borgprint(usr)
+
+		if("toggleThrusters")
+			borgo.toggle_ionpulse()
+
+		if("lampIntensity")
+			borgo.lamp_intensity = params["ref"]
+			borgo.toggle_headlamp(FALSE, TRUE)
+
+/**
+  * Forces a full update of the UI, if currently open.
+  *
+  * Forces an update that includes refreshing ui_static_data. Called by
+  * law changes and borg log additions.
+  */
+/datum/computer_file/program/borg_self_monitor/proc/force_full_update()
```

Comment:
I think theres something that does this already, either ui_update or ui_update_static_data.
Or its something on SStgui

Reply:
There's try_update_ui which requires a UI datum, which you have to get anyway. ui_update only updates ui_data I think

Reply:
Ah, there's a proc called `update_static_data` on datum, if the UI is null it does what this method does, but it doesn't have the tablet null check, so it won't work in this situation